addons/account/models/account_partial_reconcile.py

```diff
@@ -106,28 +110,34 @@ def unlink(self):
         if not self:
             return True
 
-        # Retrieve the matching number to unlink.
-        full_to_unlink = self.full_reconcile_id
-        all_reconciled = self.debit_move_id + self.credit_move_id
-
         # Retrieve the CABA entries to reverse.
         moves_to_reverse = self.env['account.move'].search([('tax_cash_basis_rec_id', 'in', self.ids)])
         # Same for the exchange difference entries.
         moves_to_reverse += self.exchange_move_id
 
+        # Retrieve the matching number to unlink
+        full_to_unlink = self.full_reconcile_id
+
+        # if the move is draft and can be removed, there is no need to update the matching number
+        all_reconciled = self.debit_move_id + self.credit_move_id
+
         # Unlink partials before doing anything else to avoid 'Record has already been deleted' due to the recursion.
         res = super().unlink()
 
         # Remove the matching numbers before reversing the moves to avoid trying to remove the full twice.
         full_to_unlink.unlink()
 
-        # Reverse CABA entries.
+        # Reverse or unlink CABA/exchange move entries.
         if moves_to_reverse:
+            not_draft_moves = moves_to_reverse.filtered(lambda m: m.state != 'draft')
+            draft_moves = moves_to_reverse.filtered(lambda m: m.state == 'draft')
+            all_reconciled = all_reconciled.filtered(lambda x: x.move_id not in draft_moves)
```

Comment:
```suggestion
            all_reconciled = all_reconciled.exists()
```
and move it after `draft_moves.unlink()`